openquake/hazardlib/tests/gsim/zhao_2016_perg_test.py

```diff
@@ -0,0 +1,195 @@
+# The Hazard Library
+# Copyright (C) 2012-2023 GEM Foundation
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import os
+import unittest
+import numpy as np
+from matplotlib import pyplot
+
+from openquake.hazardlib.geo import Point
+from openquake.hazardlib.source.rupture import get_planar
+from openquake.hazardlib.site import Site, SiteCollection
+from openquake.hazardlib.scalerel import WC1994
+from openquake.hazardlib.const import TRT
+from openquake.hazardlib.contexts import ContextMaker
+from openquake.hazardlib.gsim.zhao_2016 import ZhaoEtAl2016SSlabPErg
+from openquake.hazardlib.gsim.zhao_2016 import ZhaoEtAl2016SSlab
+
+DATA_FOLDER = os.path.join(os.path.dirname(__file__),'data', 'ZHAO16PERG',
+                           'unit_test_volc.geojson')
+
+
+def get_gms_from_ctx(imt, rup, sites, gmm_perg, gmm, azimuth):
+    """
+    Get ground-motion with and without non-ergodic path effect modifications 
+    and create trellis plots to visually inspect
+    """
+    # Create context for computation of ground-motions
+    oqp = {'imtls': {k: [] for k in [str(imt)]}, 'mags': [f'{rup.mag:.2f}']}
+    
+    # Get perg version ground-motions
+    ctxm_perg = ContextMaker(gmm_perg.DEFINED_FOR_TECTONIC_REGION_TYPE,
+                             [gmm_perg], oqp)
+    
+    ctxs_perg = list(ctxm_perg.get_ctx_iter([rup], sites)) 
+    ctxs_perg = ctxs_perg[0]
+    ctxs_perg.occurrence_rate = 0.0
+    mean_perg, std_perg, tau_perg, phi_perg = ctxm_perg.get_mean_stds(
+        [ctxs_perg])
+
+    # Get non-perg version ground-motions
+    ctxm = ContextMaker(gmm.DEFINED_FOR_TECTONIC_REGION_TYPE, [gmm], oqp)
+    ctxs = list(ctxm.get_ctx_iter([rup], sites)) 
+    ctxs = ctxs[0]
+    
+    ctxs.occurrence_rate = 0.0
```

Comment:
I believe there is no need to set the occurrence_rate

Reply:
Removed just now